provider/verify.go

```diff
@@ -179,10 +176,10 @@ func (e VerifyHandler) sendConfirmation(w http.ResponseWriter, r *http.Request)
 		Token   string
 		Site    string
 	}{
-		User:    user,
-		Address: address,
-		Token:   tkn,
-		Site:    r.URL.Query().Get("site"),
+		User:    safeHTML(user),
+		Address: safeHTML(address),
+		Token:   template.HTMLEscapeString(tkn),
+		Site:    template.HTMLEscapeString(site),
```

Comment:
vulnerability fixed: `site` comes from a URL parameter and was not being HTML escaped at all. Token is less of an issue, but I added HTML escaping just-in-case. The user and address also had some "prettifying" and truncation logic (presumably for plain-text viewing), which I have preserved in `safeHTML`, though I don't really see the point.